### ISSUE TITLE
Clean the test suite

### ISSUE THREAD

**Issue by adrienpoly** [adrienpoly/rubyvideo#5]
The test suite doesn't pass. Lots of tests are coming from the initial scaffolding. Let's make it green again by fixing incorrect tests and removing some that might be irrelevant.

At the same time, it would be nice to add a GitHub action CI file.

**Comment by Oli0li** [comment]
Hello! May I work on this please?

**Comment by adrienpoly** [comment]
Sure please do. On main I started to add the base of the CI with some linters check. Feel free to add the test to it

**Comment by adrienpoly** [comment]
@Oli0li I realised that the test suite is really messy. Not sure if you already started working on it. If not I will probably spend a bit of time fixing it in the upcoming days

**Comment by Oli0li** [comment]
Thank you for your message @adrienpoly! Yes, I've started working on it but I may have some questions for you if that's OK.

1) For tests which are erroring because there is no route for them, would you like me to remove just the tests or also the methods that are in the corresponding controllers?

I am talking about the tests for these routes:

events/new, events/create, events/destroy
speakers/new, speakers/create, speakers/destroy
talks/destroy, talks/create, talks/new

2) A lot of tests are failing because of this error: RuntimeError: Neutered Exception VCR::Errors::UnhandledHTTPRequestError
A VCR cassette exists but for some reason it is erroring. It's my first time using VCR so I would need to look more into this, I was planning on digging into this tomorrow.

3) For test/clients/youtube/channels_test.rb:6, the VCR cassette is read but it finds a different result as expected. I do not know why and would need to look into this.

4) When I was trying to fix test/controllers/speakers/enhance_controller_test.rb, I saw that the update method in that controller points to an empty file app/views/speakers/enhance/update.turbo_stream.erb, so I wasn't sure if that was normal and what to do with this.

5) You mentioned the tests should also be linted. I tried putting some redundant spaces in a test file and then ran bin/lint, and standardrb fixed it, so I think this is covered.

I have time tomorrow so I was planning on investigating all this tomorrow and let you know if I cannot figure it out. But I can also push the changes I have made so far and leave the rest to you if you prefer, just let me know what would be best for you. 🙂

**Comment by adrienpoly** [comment]
> 1. For tests which are erroring because there is no route for them, would you like me to remove just the tests or also the methods that are in the corresponding controllers?

yes they were part of the scaffold and when I removed the routes I didn't remove the tests

> 2. A lot of tests are failing because of this error: RuntimeError: Neutered Exception VCR::Errors::UnhandledHTTPRequestError

I have seen in the system test an error related to `chromedriver.storage.googleapis.com`
you can add this configuration to test_helper

```diff
VCR.configure do |c|
  c.cassette_library_dir = "test/vcr_cassettes"
  c.hook_into :webmock
+  c.ignore_hosts "chromedriver.storage.googleapis.com"
end

```
not sure it is sufficient

> 3. For test/clients/youtube/channels_test.rb:6, the VCR cassette is read but it finds a different result as expected. I do not know why and would need to look into this.

You can delete the cassette and run again the test hopefully it should fix it 

> 4. When I was trying to fix test/controllers/speakers/enhance_controller_test.rb, I saw that the update method in that controller points to an empty file app/views/speakers/enhance/update.turbo_stream.erb, so I wasn't sure if that was normal and what to do with this.

here is what I came up with 
```ruby
class Speakers::EnhanceControllerTest < ActionDispatch::IntegrationTest
  setup do
    @user = sign_in_as(users(:admin))
  end

  test "#patch" do
    patch speakers_enhance_url(speakers(:one), {format: :turbo_stream})
    assert_response :success
  end
end
```
and add an admin user to the users.yml fixture

> 5. You mentioned the tests should also be linted. I tried putting some redundant spaces in a test file and then ran bin/lint, and standardrb fixed it, so I think this is covered.

perfect 

open a PR whenever you fill there is some progress. thank you for taking this I am sorry I realize that it was quite a mess

**Comment by adrienpoly** [comment]
@Oli0li I updated the gems and fix a few things on `main` 

please rebase your branch as I think it will help you

for the VCR errors in the system test here is what fixed the error
https://github.com/adrienpoly/rubyvideo/commit/8688585e1f558fd1c12419fff2d3cf98d2e568c8

**Comment by Oli0li** [comment]
Thank you so much @adrienpoly!
I rebased and I've got your changes from [8688585](https://github.com/adrienpoly/rubyvideo/commit/8688585e1f558fd1c12419fff2d3cf98d2e568c8) in my test_helper.rb file but the clients/youtube tests using VCR are still erroring on my side. I've ran bundle install again. Is there anything else that I should do?

When I run the test suite, the only other test that is failing now apart from the VCR ones is this one:

```
EventsControllerTest#test_should_update_event:
NoMethodError: undefined method `description' for #<Event id: 980190962, date: "2017-05-01", city: "Phoenix", country_code: nil, organisation_id: 980190962, created_at: "2023-06-28 12:47:41.491956000 +0000", updated_at: "2023-06-28 12:47:41.491956000 +0000", name: "RailsConf 2017", slug: "rails-conf-2017">
    test/controllers/events_controller_test.rb:27:in `block in <class:EventsControllerTest>'
```

I saw that, as the test say, "definition", "kind" , "website" and "frequency" are attributes of "organisation", not "event", and that this form is currently not working on the website. Should I add this to the list of Issues?

There are some other tests that are failing in the test/system folder when I run them separately. I was thinking of first pushing a PR with all the things I have changed so far and start a new PR for these tests in test/system, if that's OK with you.
